### PR TITLE
[WEBRTC-535] Add a way to cancel outgoing call

### PR DESCRIPTION
**Pull request by lucaspdev**
Signed-off-by: Lucas Pais <lucasp@7g804d3.office.telnyx.com>

[WebRTC-535 - Ticket Description.](https://telnyx.atlassian.net/browse/WEBRTC-535)

---
Add a way to cancel an outgoing invitation, before an Answer is received.

### Before changes
We didn't have a way to cancel an outgoing call, nor to receive feedback on whether an invitation is in progress.

### After changes
Green call button, will turn red indicating an invitation is in progress and we can cancel it.

## ✋ Manual testing
1. Call
2. Press call button again (now will be RED)
3. Verify call is no longer in progress, and ring has stopped on the receiver.

### CODE REVIEW

File: app/src/main/java/com/telnyx/webrtc/sdk/ui/MainViewModel.kt [hunk 1]
```diff
@@ -88,8 +88,13 @@ class MainViewModel @Inject constructor(
         telnyxClient?.call?.acceptCall(callId, destinationNumber)
     }
 
-    fun endCall(callId: UUID) {
-        telnyxClient?.call?.endCall(callId)
+    fun endCall(callId: UUID? = null) {
+        callId?.let {
+            telnyxClient?.call?.endCall(callId)
+        } ?: run {
```
Comment (Oliver-Zimmerman) [inline app/src/main/java/com/telnyx/webrtc/sdk/ui/MainViewModel.kt:94]: Beautiful use of scoped functions here 👍

**Review by Oliver-Zimmerman — CHANGES_REQUESTED** [review]
Runs perfectly 👍 Can we include the empty login check as described on the ticket too?

File: app/src/main/java/com/telnyx/webrtc/sdk/ui/MainViewModel.kt (hunk 1 above)
Comment (lucaspdev) [reply app/src/main/java/com/telnyx/webrtc/sdk/ui/MainViewModel.kt:94]: Thanks! But I'm still learning Kotlin and ways to do less/more efficient code

**Comment by lucaspdev** [comment]
> Runs perfectly  Can we include the empty login check as described on the ticket too?

Yep. I'll be on that today, since I was running late for an appointment yesterday..

**Review by Oliver-Zimmerman — APPROVED** [review]

**Pull request merged**